### DISCUSSION TITLE
Adapters for the pico-ice

### DISCUSSION THREAD

**Discussion by josuah** [tinyvision-ai-inc/pico-ice#25] [Ideas]
It is planned to either offer 3rd-party purchase links or sell directly add-on for the pico-ice, such as:
* PMODs modules
* Adapter boards
* Various headers for the different soldering styles
* Maybe more...

**Comment by josuah** [comment]
Some suggestions of PMODs from the chat:
* Rotary encoder,
* 7-segment display,
* ambient light or PIR sensor
* 4 pushbuttons,
* GPS sensor,
* accelerometer, magnetosensor, gyrometer,
* wifi,
* temperature sensor
* PDM microphone
* VGA adapter
* HDMI adapter
* Ethernet adapter
* Motor driver (externally supplied): servo, brushless, 3-phase...

Some of these might be as well fit as QWIIC / Stemma QT modules through a PMOD - I²C adapter board.

**Reply by josuah** [reply]
**PMOD for SD Card**

SD Card are ubiquitous data storage useable over SPI, often present in many hobby and commercial projects.

Use-case discussed that lead to this idea: using an SD Card to store programs from which the RP2040 would start, as well as containing the FPGA bitstream.
This would allow the pico-ice to launch a FPGA bitstream + software, both contained in the SD Card, loaded onto the appropriate chip by the RP2040.

Other popular use-case, very often encountered as self-teaching project, is an MP3 file player with storage on an SD Card.

**Reply by nobodywasishere** [reply]
a PDM microphone and a vga adapter would be nice, may design ones myself soon.

**Reply by josuah** [reply]
Great idea, a lot of connectors for a lot of different connectors. That makes sense.

**Comment by josuah** [comment]
The power shield project suggested by @z80

**Comment by josuah** [comment]
Some suggestion of adapter shields from the chat:

* Adapter for breadboards: ribbon cable to plug onto the breadboard
* Adapter for breadboards: double PMOD to single-row header cable
* Adapter for QWIIC / Stemma QT / ... connector system over I²C
* Adapter holding the board in place, giving power supply to a breadboard, exposing PMODs as vertical female headers

**Comment by josuah** [comment]
Suggestions of headers to solder from the chat:

* The classic angled 2-rows PMOD headers
* 2-pin female header for use with the BOOTSEL button along with a jumper
* full width, 2-rows female headers to cover all pins
* single-row female headers, in addition with single-row male headers, to have the inner rows on a breadboard, and the outter rows with female headers faced up
